tools/regression_test.py

```diff
@@ -0,0 +1,980 @@
+            # mmseg
+            metric = eval(metric_str.strip().split('|')[2])
+            if metric <= 1:
+                metric *= 100
+        elif yaml_metric_key in ['AP', 'AR']:
+            # mmpose
+            metric = eval(metric_str.split(': ')[-1])
+        elif yaml_metric_key == '0_word_acc_ignore_case' or \
+                yaml_metric_key == '0_hmean-iou:hmean':
+            # mmocr
+            evaluate_result = eval(metric_str)
+            if not isinstance(evaluate_result, dict):
+                print(f'Got error metric_dict = {metric_str}')
+                return 'x'
+            metric = evaluate_result.get(yaml_metric_key, 0.00)
+            if yaml_metric_key == '0_word_acc_ignore_case':
+                metric *= 100
+        elif yaml_metric_key in ['Eval-PSNR', 'Eval-SSIM']:
+            # mmedit
+            metric = eval(metric_str.split(': ')[-1])
+        else:
+            metric = 'x'
+        info_value = metric
+    else:
+        info_value = 'x'
+
+    return info_value
+
+
+def compare_metric(metric_value, metric_name, pytorch_metric, metric_info):
+    """Compare metric value with the pytorch metric value and the tolerance.
+
+    Args:
+        metric_value (float): Metric value
+        metric_name (str): metric name.
+        pytorch_metric (dict): Pytorch metric which get from metafile.
+        metric_info (dict): Metric info from test yaml.
+
+    Returns:
+        Bool: If the test pass or not.
+    """
+    if metric_value == 'x':
+        return False
+
+    metric_pytorch = pytorch_metric.get(str(metric_name))
+    tolerance_value = metric_info.get(metric_name, {}).get('tolerance', 0.00)
+    if (metric_value - tolerance_value) <= metric_pytorch <= \
+            (metric_value + tolerance_value):
+        test_pass = True
+    else:
+        test_pass = False
+    return test_pass
+
+
+def get_fps_metric(shell_res, pytorch_metric, metric_key,
+                   yaml_metric_info_name, log_path, metrics_eval_list,
+                   metric_info, logger):
+    """Get fps and metric.
+
+    Args:
+        shell_res (int): Backend convert result: 0 is success
+        pytorch_metric (dict): Metric info of pytorch metafile
+        metric_key (str):Metric info.
+        yaml_metric_info_name (str): Name of metric info in test yaml.
+        log_path (str): Logger path.
+        metrics_eval_list (dict): Metric list from test yaml.
+        metric_info (dict): Metric info.
+        logger (Logger): Logger handler.
+
+    Returns:
+        Float: fps: FPS of the model
+        List: metric_list: metric result list
+        Bool: test_pass: If the test pass or not.
+    """
+    metric_list = []
+
+    # check if converted successes or not.
+    if shell_res != 0:
+        fps = 'x'
+        metric_value = 'x'
+    else:
+        # Got fps from log file
+        fps = get_info_from_log_file('FPS', log_path, metric_key)
+        print(f'Got fps = {fps}')
```

Comment:
Why don't we use logger here?

Reply:
Done. I replace all `print` to `logger.xxx`